dts/bindings/clock/st,stm32-clock-mco.yaml

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2024, Joakim Andersson
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+compatible: "st,stm32-clock-mco"
+
+description: |
+        STM32 Microcontroller clock output (MCO)
+        Configure the microcontroller clock output pin and additional properties,
+        e.g. clock source and prescaler.
+        Example:
+                &mco {
+                        source = <STM32_SRC_LSE>;
+                        prescaler = <2>;
```

Comment:
Binding-wise, I have a preference to reuse the same binding as the usual stm32 `clocks` binding, as for instance: 
```
clocks = <&rcc STM32_SRC_PLL_Q MCO_SEL(4)>,
```
See [my comment here](https://github.com/zephyrproject-rtos/zephyr/pull/68846#discussion_r1502800920).

As an additional benefit, it would remove the whole `#if DT_PROP(DT_NODELABEL(mco1), source) == NO_SEL` section at the beginning of this file and avoids to face LL API discrepancy issues (like the `#if (CONFIG_SOC_SERIES_STM32G0X || ...` above).

The only trick would that we wouldn't be allowed to use clock_control API as MCO clocks are supposed to be configured before PLLs (as mentioned in a previous comment).

This being said, I won't block for this.

Reply:
By convention, the `xxx_SEL()` macros accept the value that goes in mux selection register (corresponding here to the `LL_RCC_MCOxSOURCE_XXX` definitions).

In peripherals, it also encodes RCC offset, mask and position, but it wouldn't (shouldn't?) be necessary in this case.

Reply:
I'm not sure what you mean by register offset.
But at the moment the STM32_CLOCK macro only allows 3-bit values, and the MCO selection is 4-bits.

If MCO_SEL(4) is the input to the conf register then chooses the clock, so why do we need STM32_SRC_PLL_Q  in addition?
We have 0100: HSE clock selected, then having STM32_SRC_HSE in addition means there is two values that must match.

What is recommendation here then?
Still define MCO_SEL and use raw numbers in this?

Reply:
@erwango 
```
#define MCO_SEL_HSE 4
#define MCO_PRE_DIV_4 2
#define MCO_SEL(val) ((val & 0xf) << 24)
#define MCO_PRE(val) ((val & 0x3) << 28)

source = <MCO_SEL(MCO_SEL_HSE)>
prescaler = <MCO_PRE(MCO_PRE_DIV_4)>
```
Seems to be the most clear to me, the user can then easily see that the have chosen HSE divided by 4.

Reply:
That would work for me, thanks.

Reply:
Updated approach after discord discussion.

Will use something like:
```
clocks = <&rcc STM32_SRC_PLL_Q MCO_SEL(4)>,
prescaler = <MCO_PRE(2)>;
```
Where MCO_SEL and MCO_PRE encodes values that go directly into the RCC_CFGR register.
Using STM_CLOCK_CFGR macro because of bit-width requirement for these values.